node/src/proxy_server/http_protocol_pack.rs

```diff
@@ -67,12 +67,15 @@ impl HttpProtocolPack {
         match parts.len() {
             1 => Some(Host {
                 name: parts.remove(0).to_string(),
-                port: None,
-            }),
-            2 => Some(Host {
-                name: parts.remove(0).to_string(),
-                port: Self::port_from_string(parts.remove(0).to_string()),
+                port: HTTP_PORT,
```

Comment:
Now, instead of making a missing port number `None` here in `HttpProtocolPack`, where we know missing port numbers mean a default of 80, and then figuring out the number later when we run into the `None` outside the HTTP context, we're ditching the `Option` and defaulting it to 80 right here.